src/app/children/children.service.ts

```diff
@@ -80,7 +99,72 @@ export class ChildrenService {
     return this.db.saveDatabaseIndex(designDoc);
   }
 
+  private createChildSchoolRelationIndex(): Promise<any> {
+
+    const designDoc = {
+      _id: '_design/childSchoolRelations_index',
+      views: {
+        by_child: {
+          map: `(doc) => {
+            if (!doc._id.startsWith("${ChildSchoolRelation.ENTITY_TYPE}")) return;
+            emit(doc.childId);
+            }`
+        },
+        by_school: {
+          map: `(doc) => {
+            if (!doc._id.startsWith("${ChildSchoolRelation.ENTITY_TYPE}")) return;
+            if (doc.end) return;
+            emit(doc.schoolId);
+            }`
+        },
+        by_date: {
+          map: `(doc) => {
+            if (!doc._id.startsWith("${ChildSchoolRelation.ENTITY_TYPE}")) return;
+            emit(doc.childId + (new Date(doc.start)).getTime());
```

Comment:
maybe for readability during debugging add a separator character: `emit(doc.childId + '_' + (new Date(doc.start)).getTime());`